### PR TITLE
influxdb version 0.9 support

### PR DESCRIPTION
**Pull request by imreACTmd**
Signed-off-by: imre Fitos imre@act.md

### CODE REVIEW

**Comment by imreACTmd** [comment]
Sure, if that's what holding it back from accepting the pull request I will replace the elif with an else, just let me know.

File: src/diamond/handler/influxdbHandler.py [hunk 1]
```diff
@@ -120,6 +123,7 @@ def get_default_config(self):
             'batch_size': 1,
             'cache_size': 20000,
             'time_precision': 's',
+            'influxdb_version': 0.8,
```
Comment (mattrobenolt) [inline src/diamond/handler/influxdbHandler.py]: Shouldn't this be a string? Version numbers aren't floats.

**Comment by mattrobenolt** [comment]
I agree with @ctrlrsf, I don't want to be revisiting this when 0.10 or something is released. So instead of passing along version, what about just a boolean for old/new behavior?

**Comment by ctrlrsf** [comment]
@mattrobenolt I checked with folks on Influx IRC channel and turns out API could still continue to evolve in future versions (duh). Boolean I think makes sense now, but will we then end up with `influx_compat_v08` and `influx_compat_v09` variables if there's another breaking change?

Another alternative was maybe calling it `influx_compat_mode` and it could then be set to "v08" now to support 0.8.x server, and left unset to support latest (0.9.x). Same variable could later be set to "v09" to support 0.9.x server if 0.10.0 changes something else.

**Comment by Investocat** [comment]
There isn't a way to query the api version from the server itself?

**Comment by imreACTmd** [comment]
Looks like there is an X-Influxdb-Version header in 0.8 and 0.9, I'll see what I can do.

**Comment by airtoxin** [comment]
It needs to change import statement because currently influxdb-python modules uses 0.9.x api based client by default.

```python
 try:
    if self.influxdb_version == '0.8':
         from influxdb.influxdb08 import InfluxDBClient
    else:
         from influxdb import InfluxDBClient
 except ImportError:
     InfluxDBClient = None
```

**Comment by jwestboston** [comment]
+1

**Comment by justin8** [comment]
This doesn't seem to work for me? The latest influxdb lib apparently needs to import influxdb.influxdb08 for compatibility since the api changed. but I can't see how this code would handle that at all. Perhaps after establishing the desired version, import the correct version of InfluxDBClient?

**Comment by imreACTmd** [comment]
I will have an updated version soon! Waiting for the 0.9 API to stabilize, they just introduced a new plaintext API as well.

**Comment by justin8** [comment]
Good plan. I'll stay subscribed here then. Thanks

**Comment by jwestboston** [comment]
Looking forward to this now that InfluxDB 0.9.0 is released as stable.

**Comment by slnowak** [comment]
What's the status of this PR? Could you merge it?

**Comment by imreACTmd** [comment]
I'm back from vacation and will be updating it soon (unless someone else wants to do it)

**Comment by jwestboston** [comment]
Merge merge merge :-)

**Comment by Ormod** [comment]
Just a FYI In order to get legible results from Diamond through to Influxdb 0.9.3 (influxdb-python 2.8) and from there to view them reasonably in Grafana 2.1 I had to also add this commit: https://github.com/Ormod/Diamond/commit/dbe971aaf3df8809def408468928bae0893eaa65

**Comment by rhunwicks** [comment]
We're waiting for this PR to merge before deploying Diamond.

File: src/diamond/handler/influxdbHandler.py [hunk 2]
```diff
@@ -170,11 +174,19 @@ def _send(self):
...
+                    if self.influxdb_version == "0.8":
+                        for path in self.batch:
+                            metrics.append({
+                                "points": self.batch[path],
+                                "name": path,
+                                "columns": ["time", "value"]})
+                    elif self.influxdb_version == "0.9":
+                        for path in self.batch:
+                            metrics.append({
+                                "name": path,
```
Comment (peggyl) [inline src/diamond/handler/influxdbHandler.py]: This key name should be "measurement", not "name".

This is required to be consistent with influxdb-client: https://github.com/influxdb/influxdb-python/blob/efdac997faeb36984fa4c949e1c57ca5a6e1303d/influxdb/line_protocol.py#L94

**Comment by jaingaurav** [comment]
@imreACTmd Any chance you can resist this PR and address the comments?

**Comment by imreACTmd** [comment]
Hi Gaurav, I will definitely try!

**Comment by dibits** [comment]
Tested it with the recommended changes and it works like a charm. If @imreACTmd can't do it soon I'll send another pull request with the changes.

**Comment by shortdudey123** [comment]
Will be superseded by https://github.com/python-diamond/Diamond/pull/504

**Pull request closed without merging**